project1/thread_pool/thread_pool_interface.h

```diff
@@ -0,0 +1,41 @@
+#ifndef PROJECT1_THREAD_POOL_INTERFACE_H
+#define PROJECT1_THREAD_POOL_INTERFACE_H
+
+#include <memory>
+
+#include "task.h"
+
+namespace thread_pool {
+
+/**
+ * @brief Standardized interface for thread pools.
+ */
+class IThreadPool {
```

Comment:
I don't see a `RunTask` function or equivalent. I'm not experienced with threads, so I'm not exactly sure how a task will run.

Reply:
See my comment above about the `RunAtomic` method in `Task`.

In this case, a new task is submitted to the pool by calling `AddTask`. This will automatically run the task as soon as possible.